packages/angular_devkit/build_angular/src/webpack/configs/styles.ts

```diff
@@ -419,3 +373,69 @@ function getTailwindConfigPath({ projectRoot, root }: WebpackConfigOptions): str
 
   return undefined;
 }
+
+function getSassLoaderOptions(
+  root: string,
+  projectRoot: string,
+  implementation: SassWorkerImplementation | SassLegacyWorkerImplementation,
+  includePaths: string[],
+  indentedSyntax: boolean,
+  verbose: boolean,
+): Record<string, unknown> {
+  return implementation instanceof SassWorkerImplementation
+    ? {
+        sourceMap: true,
+        api: 'modern',
+        implementation,
+        // Webpack importer is only implemented in the legacy API.
+        // See: https://github.com/webpack-contrib/sass-loader/blob/997f3eb41d86dd00d5fa49c395a1aeb41573108c/src/utils.js#L642-L651
+        webpackImporter: false,
+        sassOptions: {
+          loadPaths: [
+            ...includePaths,
+            // Needed to resolve node packages and retain the same behaviour of with the legacy API as sass-loader resolves
+            // scss also from the cwd and project root.
+            // See: https://github.com/webpack-contrib/sass-loader/blob/997f3eb41d86dd00d5fa49c395a1aeb41573108c/src/utils.js#L307
+            projectRoot,
+            path.join(root, 'node_modules'),
```

Comment:
This workaround won't support package `exports` fields nor Yarn PnP.  We'll need to do a followup with a fix for this.  Probably a custom Sass importer that tries node resolution first from the project root.

Reply:
Yes, I already have some ideas for that.